dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilder.java

```diff
@@ -86,6 +89,35 @@ private static DependencyNode resolveCompileTimeDependencies(Artifact artifact)
     return node;
   }
 
+  private static DependencyNode resolveCompileTimeDependencies(List<Artifact> artifacts)
+      throws DependencyCollectionException, DependencyResolutionException {
+    RepositorySystemSession session = RepositoryUtility.newSession(system);
```

Comment:
move this closer to where it's first needed so it isn't constructed if this returns early from the cache

Reply:
Done